[deberinoo] iP

deborahluyx opened the pull request:
# 🚀 Juno: Task Manager
> *"One small step at a time!"* - Juno

Juno isn't just a task manager. She's your **mission control**, keeping track of your deadlines, to-dos, and events with **precision and efficiency**.

## ⚡ Why Juno?  
- 🔹 **Minimal & Efficient** – No fluff, just what you need.  
- 🔹 **Command-Based Control** – Type it, execute it, done.  
- 🔹 **Blazing Fast** – No delays. No hesitation. Just action.  

## 📌 Your Mission, Should You Choose to Accept It:  
1. **[Download Juno](https://github.com/deberinoo/ip)** – Secure the latest build.  
2. **Deploy Juno** – Launch the program by double-clicking.  
3. **Execute Commands** – Add, mark, and track your tasks like a true strategist.  
4. **Dominate Your Schedule** – Juno’s got your back. Always.  

## 🛠 Tactical Features  
- [x] ✅ **Task Management** – Keep things in order.  
- [ ] ⏳ **Deadline Tracking** – _(Mission update coming soon)_  
- [ ] ⏰ **Automated Reminders** – _(In development by Juno’s engineering team)_  

## 🖥️ Systems Check  
If you're a **Java developer**, you can even tweak Juno to your liking! Here’s her `main` system boot sequence:  

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

deborahluyx committed:
level 0: rename, greet, exit

deborahluyx committed:
level 1: echo

deborahluyx committed:
level 2: add, list

deborahluyx committed:
level 3: mark as done

deborahluyx committed:
level 4: todo, event, deadline

deborahluyx committed:
a-textuitesting

deborahluyx committed:
level 5: handle errors

deborahluyx committed:
Update README.md

deborahluyx committed:
level 6: delete

deborahluyx committed:
Update README.md

deborahluyx committed:
add enums

deborahluyx committed:
Merge branch 'master' of https://github.com/deberinoo/ip

deborahluyx committed:
Implement task storage with file loading and saving functionality

deborahluyx committed:
Merge branch 'branch-Level-7'

deborahluyx committed:
Add date and time handling for tasks

deborahluyx committed:
Merge branch 'branch-Level-8'

deborahluyx committed:
Refactor: Update task date handling and improve task display logic

deborahluyx committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

deborahluyx committed:
Set up Gradle build configuration and configure for testing with JUnit

deborahluyx committed:
Add JavaDoc

deborahluyx committed:
Merge branch 'branch-A-JavaDoc'

deborahluyx committed:
Add missing JavaDoc for TasksOnCommand.java

deborahluyx committed:
Remove redundant code (isDone as parameter)

deborahluyx committed:
Organise imports and list imported classes explicitly

deborahluyx committed:
Merge branch 'branch-A-CodingStandard'

deborahluyx committed:
Add find task feature and JavaDoc

deborahluyx committed:
Merge branch 'branch-Level-9'

deborahluyx committed:
Add checkstyle, suppression

deborahluyx committed:
Merge branch 'branch-A-CheckStyle'

deborahluyx committed:
Add GUI

deborahluyx committed:
Merge branch 'branch-Level-10'

bryantjandra commented on src/main/java/juno/TaskList.java:
```diff
@@ -0,0 +1,140 @@
+    public Task getTask(int index) {
+        // if (index < 0 || index >= tasks.size()) {
+        //     throw new JunoException("Invalid task index.");
+        // }
```
This commented-out check suggests an alternative error-handling approach. If it isn't intended for use, should this perhaps be removed or finalized?

bryantjandra commented on src/main/java/juno/Ui.java:
```diff
@@ -0,0 +1,210 @@
+     * 
+     * @param matchingTasks The list of tasks that match the search keyword.
+     */
+    public void showMatchingTasks(List<Task> matchingTasks) {
```
Many of the Ui class methods have short, clear names like showGoodbye(), showError(...), etc., which is great! I like how you’ve kept them descriptive (e.g., showLoadingError(), showMatchingTasks()) as well.

bryantjandra commented on src/main/java/juno/Parser.java:
```diff
@@ -0,0 +1,110 @@
+     * @throws JunoException If the input does not match any recognized command.
+     */
+    public static Command parse(String input) throws JunoException {
+        if (input.equalsIgnoreCase("bye")) {
```
Have you considered using a switch statement instead of chaining if-else blocks? While not strictly required, the parseTask function below already uses a switch, so maintaining consistency across similar logic structures might improve readability.

bryantjandra commented on src/main/java/juno/Parser.java:
```diff
@@ -0,0 +1,110 @@
+            DateTimeFormatter.ofPattern("MMM dd yyyy")  // Format like Jan 31 2025
+        };
+
+        switch (type) {
+            case "T":
+                return new TodoTask(description);
+            case "D":
+                LocalDate deadline = parseDate(parts[3], dateFormatters);
+                return new DeadlineTask(description, deadline);
+            case "E":
+                LocalDate eventStart = parseDate(parts[3], dateFormatters);
+                LocalDate eventEnd = parseDate(parts[4], dateFormatters);
+                return new EventTask(description, eventStart, eventEnd);
+            default:
+                throw new IllegalArgumentException("Invalid task type in saved data.");
+        }
```
Shouldn't this switch statement not be indented? This is because the `switch` statement should follow the given coding standard, where `case` clauses are not indented. You might want to configure your IDE to enforce this style automatically.

bryantjandra commented on src/main/java/juno/javafx/DialogBox.java:
```diff
@@ -0,0 +1,60 @@
+    }
+
+    public static DialogBox getJunoDialog(String text, Image img) {
+        var db = new DialogBox(text, img);
+        db.flip();
+        return db;
```
Great job using a short, descriptive name `flip()`. The method name is a clear verb. Also, the local variable `db` is short, but it’s in a small scope, so that’s aligned with “variables with a small scope can have short names.” Looks good under the naming rules!

broyantt reviewed (commented):
LGTM from a coding-standard perspective. Just a few nits as mentioned! Overall, I found your code easy to read and understand. Keep up the great work!

meloppeitreet commented on src/main/java/Juno.class:
Class files should be excluded from commit (.gitignore)

meloppeitreet reviewed (commented):
Code naming standards LGTM overall 👍

deborahluyx committed:
Refactor command classes, integrate task management, and clean up Javadocs

Refactor command classes, integrate task management with UI, and clean up Javadocs

Major refactoring of command classes to reduce code repetition and remove
unnecessary code. This improves maintainability and readability of the
codebase.

Integrated task management functionality with the UI to provide a seamless
user experience. This allows users to manage tasks directly through the
interface.

Organised files into respective folders for better structure and clarity.
This change ensures that related files are grouped logically, making it
easier to navigate the project.

Removed existing Javadocs due to their length and unnecessary verbosity.
Javadocs will be replaced in a future commit with concise and relevant
documentation.

deborahluyx committed:
Add assertions

deborahluyx committed:
Merge pull request #1 from deberinoo/branch-A-Assertions

Add assertions

deborahluyx committed:
Improve code quality: Rename variables for clarity

deborahluyx committed:
Merge pull request #2 from deberinoo/branch-A-CodeQuality

Improve code quality: Rename variables for clarity

deborahluyx committed:
Add JavaDoc

deborahluyx committed:
Add streams

deborahluyx committed:
Add HelpCommand to display available commands

deborahluyx committed:
Merge pull request #5 from deberinoo/branch-A-Streams

Add streams

deborahluyx committed:
Merge pull request #6 from deberinoo/branch-A-Extension

Add HelpCommand to display available commands

deborahluyx committed:
Improve Gui

deborahluyx committed:
Merge pull request #8 from deberinoo/branch-A-BetterGui

Improve Gui

deborahluyx committed:
Delete redundant code, rename Duke to Juno

deborahluyx committed:
Edit header comment

deborahluyx committed:
Add representative screenshot

deborahluyx committed:
Edit Readme

deborahluyx committed:
Update README.md

deborahluyx committed:
Delete Ui.PNG

deborahluyx committed:
Merge branch 'master' of https://github.com/deberinoo/ip

deborahluyx committed:
Add Ui.png

deborahluyx committed:
Update README.md

deborahluyx committed:
Update README.md
